mvpa2/misc/plot/scatter.py

```diff
@@ -0,0 +1,643 @@
+      should only have 2 items
+    mask: array, optional
+      Additional mask to specify which values do not consider to plot.
+      By default values with 0s in both dimensions are not plotted.
+    masked_opacity: float, optional
+      By default masked out values are not plotted at all.  Value in
+      (0,1] will make them visible with this specified opacity
+    labels: list of str, optional
+      Labels to place for x and y axes
+    colors: bool or string or colormap, optional
+      Either to use colors to associate with physical location and
+      what colormap to use (jet by default if colors=True)
+    dimcolor: int
+      If `colors`, then which dimension (within given 3D volume) to
+      "track"
+    limits: 'auto', 'same', 'per-axis' or (min, max)
+      Limits for axes: when 'auto' if data ranges overlap is more than
+      50% of the union range, 'same' is considered.  When 'same' --
+      the same limits on both axes as determined by data.  If
+      two-element tuple or list is provided, then that range is
+      applied to both axes.
+    hint_opacity: float, optional
+      If `colors` is True, to then a "slice" of the volumetric data
+      is plotted in the specified opacity to hint about the location
+      of points in the original Xd data in `dimcolor` dimension
+    x_jitter: float, optional
+      Half-width of uniform noise added to x values.  Might be useful if data
+      is quantized so it is valuable to jitter points a bit.
+    y_jitter: float, optional
+      Half-width of uniform noise added to y values.  Might be useful if data
+      is quantized so it is valuable to jitter points a bit
+    fig : Figure, optional
+      Figure to plot on, otherwise new one created
+    ax_*: axes, optional
+      Axes for the scatter plot and histograms. If none of them is specified
+      (which is the default) then 'classical' plot is rendered with histograms
+      above and to the right
+    bp_location: ('scatter', 'hist', None), optional
+      Where to place boxplots depicting data range
+    xlim: tuple, optional
+    ylim: tuple, optional
+      To fix plotted range
+    rasterized: bool, optional
+      Passed to scatter call, to allow rasterization of heavy scatter plots
+    uniq: bool, optional
+      Plot uniq values (those present in one but not in the other) along each axis
+      with crosses
+    """
+    if len(dataXd) != 2:
+        raise ValueError("First axis of dataXd can only have two dimensions, "
+                         "got {0}".format(len(dataXd)))
+    dataXd = np.asanyarray(dataXd)      # TODO: allow to operate on list of arrays to not waste RAM/cycles
+    data = dataXd.reshape((2, -1))
+    if dataXd.ndim < 5:
+        ntimepoints = 1
+    elif dataXd.ndim == 5:
+        ntimepoints = dataXd.shape[-1]
+    else:
+        raise ValueError("Do not know how to handle data with %d dimensions" % (dataXd.ndim - 1))
+    if x_jitter or y_jitter:
```

Comment:
just smoke test this one please (the same invocation but with x_jitter=0.1 or so)